pkg/chunk/cassandra/storage_client.go

```diff
@@ -70,6 +66,14 @@ func (cfg *Config) session() (*gocql.Session, error) {
 	cluster.ConnectTimeout = cfg.ConnectTimeout
 	cfg.setClusterConfig(cluster)
 
+	session, err := cluster.CreateSession()
+	if err == nil || err != gocql.ErrNoConnectionsStarted {
+		return session, nil
+	}
+	// keyspace not exist
+	if err := cfg.createKeyspace(); err != nil {
```

Comment:
Isn’t the logic wrong now? This will call `createKeyspace` even if it there was no error.

Reply:
```go
if err == nil || err != gocql.ErrNoConnectionsStarted {
	return session, nil
}
```
If `no error` or `error is not keyspace not exist` then just return session and error.
So rest part is only for keyspace not exist.

Reply:
Correct, my mistake.